### PR TITLE
[sentry-native] Fix linux runtime error and windows usage

### LINKED ISSUE #39114: sentry-native port requires zlib dependency on windows + problems on linux
**Describe the bug**
Hi!
I have two issues regarding the using of the sentry-native port.
1. While trying to generate the cmake project build configuration while using `sentry-native` as vcpkg dependency, i'm getting this error:
```
1> [CMake] CMake Error at out/build/Windows-x86-Debug/vcpkg_installed/x86-windows/share/sentry/sentry_crashpad-targets.cmake:105 (set_target_properties):
1> [CMake]   The link interface of target "sentry_crashpad::zlib" contains:
1> [CMake] 
1> [CMake]     ZLIB::ZLIB
```
This problem might be solved by adding additional project dependency for zlib in vcpkg, however i'm not sure if this is correct behaviour, shouldn't `sentry-native` be a standalone port?

2. On linux the cmake project generation & building it works fine, however while trying to run the app with dummy crash (assigning a `nullptr` after initialization of sentry) the crash raport doesn't gets send to the sentry backend.
```
[1717:1717:20240604,064748.491454:FATAL pthread_create_linux.cc:54] Check failed: next_pthread_create. dlsym: RTLD_NEXT used in code not dynamically loaded
```

**Environment**
- OS: Windows (x86)/Linux (x64)

**To Reproduce**
Steps to reproduce the behavior (linux):
1. Add `VCPKG_ROOT` path to locally installed vcpkg
2, Generate a project via this command: `cmake . --preset Linux-x64-Release`
3. Compile the project via this command `cmake --build . --preset Linux-x64-Release`
4. Copy the `crashpad-handler` executable from `out/build/Linux-x64-Release/vcpkg_installed/x64-linux/tools/sentry-native` into `/out/build/Linux-x64-Release`

**Expected behavior**
The application should send a crashreport to a sentry backend on linux.  
I'm not getting any issues while building & running my other project that depends on `sentry-native` as source dependency (crash report is being properly send to sentry backend).

**Additional context**
I'm using sentry-native 0.7.5 version (baseline: 67cc1677c3bf5c23ea14b9d2416c7422fdeac492)

And  here's my ready to compile project that you can use to reproduce the issue
[cmake-test.zip](https://github.com/user-attachments/files/15542706/cmake-test.zip)

You just need to set the `DSN` in `main.cpp`, and you're ready to build/test the project.

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #39114

1. Fix  sentry-native:x64-windows-static usage
2. Apply upstream change fix sentry-native linux runtime error. related issue: https://github.com/getsentry/sentry-native/issues/596#issuecomment-944118365

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/sentry-native/fix-crashpad-pthread.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@@ -89,12 +89,6 @@ if(NOT IOS)
+         main.cc
+     )
+ 
+-    if(LINUX)
+-        target_sources(crashpad_handler PRIVATE
+-            ../client/pthread_create_linux.cc
+-        )
+-    endif()
+-
```
Comment (dg0yt) [inline ports/sentry-native/fix-crashpad-pthread.patch:14]: IMO this change is incorrect. Cf. second part of https://github.com/microsoft/vcpkg/issues/39114#issuecomment-2146636567

**Review by dg0yt — CHANGES_REQUESTED** [review]
> [sentry-native] Fix linux build error

AFAICT there is no build error. There is a *runtime* error.
And that's why CI won't sufficiently indicate that the change is okay.

**Comment by JonLiu1993** [comment]
Tested usage successfully by `sentry-native:x64-windows-static`

**Review by WangWeiLin-MV — APPROVED** [review]
The port usage tests pass with the following triplets:
* x64-linux
* x64-windows-static

**Comment by dg0yt** [comment]
The changes should also be upstreamed, to have them included in future releases.

**Comment by JonLiu1993** [comment]
I have commit PR for upstream : 
https://github.com/getsentry/sentry-native/pull/1005 
https://github.com/getsentry/crashpad/pull/103

**Pull request merged**